### ISSUE TITLE
The last two lines of doc comments are always missing in popup window.

### ISSUE THREAD

**Issue by your-diary** [fannheyward/coc-rust-analyzer#1221]
I have this Rust code:

```rust
/// 1
/// 2
/// 3
/// 4
/// 5
struct S;

fn main() {
    S {};
}
```

The doc comments of the struct `S` consist of five lines.

However, when I try to show the documentation of `S` via `:call CocAction('doHover')`, the last two lines of the doc comments are missing:

![Screenshot 2024-01-21 at 19 54 50](https://github.com/fannheyward/coc-rust-analyzer/assets/58146437/d8dfefd3-5e47-45c8-829e-72d2b5a4dd9e)

This is extremely annoying especially when doc comments consist of one or two comments:

![Screenshot 2024-01-21 at 20 08 03](https://github.com/fannheyward/coc-rust-analyzer/assets/58146437/63e58590-61a9-464b-a214-b48a097a4433)

Though I can scroll with a mouse wheel, I think the entire doc comments should be shown (if they are short enough as in the example above).

By the way, `rust-analyzer` extension of VSCode shows the doc comments correctly.

![Screenshot 2024-01-21 at 19 55 10](https://github.com/fannheyward/coc-rust-analyzer/assets/58146437/b242df48-bc67-4650-824b-a30288a78077)

And Golang has no problem:

![Screenshot 2024-01-21 at 20 14 21](https://github.com/fannheyward/coc-rust-analyzer/assets/58146437/d91dc7dc-15c2-4a61-a944-90e413dcaac7)

**What's the output of `:CocInfo`**

```
## versions

vim version: VIM - Vi IMproved 9.1 9010000
node version: v21.6.0
coc.nvim version: 0.0.82-d1568d56 2023-09-29 19:43:34 +0800
coc.nvim directory: /Users/user/.vim/plugged/coc.nvim
term: iTerm.app
platform: darwin

## Log of coc.nvim

2024-01-21T19:53:20.977 INFO (pid:21892) [plugin] - coc.nvim initialized with node: v21.6.0 after 126
2024-01-21T19:53:20.978 INFO (pid:21892) [services] - LanguageClient Rust Analyzer Language Server state change: stopped => starting
2024-01-21T19:53:20.982 INFO (pid:21892) [language-client-index] - Language server "rust-analyzer" started with 21894
2024-01-21T19:53:20.987 INFO (pid:21892) [services] - LanguageClient Rust Analyzer Language Server state change: starting => running
2024-01-21T19:53:21.004 INFO (pid:21892) [services] - service rust-analyzer started
2024-01-21T19:53:22.188 INFO (pid:21892) [attach] - Request action: doHover []
```

**What's the output of `:CocCommand rust-analyzer.serverVersion`**

```
[coc.nvim] rust-analyzer 0.3.1807-standalone (9d8889cdf 2024-01-11)
```

**What's your coc-rust-analyzer version? You can get it from `:CocList extensions`**

```
* coc-rust-analyzer 0.75.0 ~/.config/coc/extensions/node_modules/coc-rust-analyzer
```

**Comment by fannheyward** [comment]
<img width="634" alt="image" src="https://github.com/fannheyward/coc-rust-analyzer/assets/345274/d60bedbb-a34c-4e6a-8685-3afde12db685">

Can't reproduce this.

**Comment by your-diary** [comment]
@fannheyward 
Thank you for trying that.

Reading your response, I tried to reproduce the behavior with minimal configurations. Here are the minimal reproducible configurations:

`~/.vimrc`

```vimrc
call plug#begin('~/.vim/plugged')
    Plug 'neoclide/coc.nvim', {'branch': 'release'}
call plug#end()

let g:coc_global_extensions = ['coc-rust-analyzer']
```

`:CocConfig`

```json
{}
```

`~/.vim/syntax/rust.vim`

```vimrc
set number
```

**Comment by fannheyward** [comment]
I also have `set number` in my vimrc, will test this with syntax's.

**Comment by your-diary** [comment]
If I remove `set number` from `~/.vim/syntax/rust.vim` and instead put `autocmd BufRead *.rs set number` in `~/.vimrc`, the issue disappears.
This is very strange.

**Comment by fannheyward** [comment]
The floating doc buffer uses the `syntax/rust.vim` setting, display the number line in floating window, decrease the width by `numberwidth`.

The problem: why the floating buffer uses `syntax/rust.vim` cause its filetype is not rust.

**Comment by your-diary** [comment]
I see...

> why the floating buffer uses syntax/rust.vim cause its filetype is not rust.

To highlight the contents inside it, perhaps?

It seems the behavior is not limited to Rust. I tried Go:

```vimrc
" ~/.vim/syntax/go.vim

syntax match __red /./
highlight __red cterm=bold,italic ctermfg=red

set number
```

![Screenshot 2024-01-23 at 21 32 27](https://github.com/fannheyward/coc-rust-analyzer/assets/58146437/fff6467e-eeaa-4fe6-b040-f54f46941b40)

Reading your comment, I noticed the line numbers are also shown in the screenshot of Go though no line is missing.
